src/parse.js

```diff
@@ -128,8 +180,8 @@ function parse(source, root, options) {
         } catch (e) {
 
             /* istanbul ignore else */
-            if (acceptTypeRef && typeRefRe.test(token))
-                return token;
+            if (acceptIdentifier && nameRe.test(token))
+                return readIdentifier(token); // `ENUM_VALUE`
 
             /* istanbul ignore next */
             throw illegal(token, "value");
```

Comment:
I think readRanges needs to be updated as well.  In edition 2023 we changed the grammar of reserved field/enum value names to be identifiers instead of strings.  This avoids accidental concatenation